File: 151-reverse-words-in-a-string/reverse-words-in-a-string.cpp
class Solution {
public:
    string reverseWords(string s) {
        string ans = "";
        vector<string> str;
        string temp = ""; // Moved outside the loop
        for (int i = 0; i < s.size(); i++) {
            if (isspace(s[i]) && temp.size() != 0) {
                str.push_back(temp);
                temp = ""; // Reset temp for the next word
            } else if (!isspace(s[i])) {
                temp += s[i];
            }
        }

        // Add the last word if not empty
        if (temp.size() != 0) {
            str.push_back(temp);
        }
        cout << str.size() << endl;

        int start=0;
        int end=str.size()-1;
        while(start<=end){
            swap(str[start],str[end]);
            start++;
            end--;
        }

        for(auto s:str){
            ans+=s;
            ans+=" ";
        }
        ans.pop_back();
        return ans;
    }
};

File: 151-reverse-words-in-a-string/reverse-words-in-a-string.js
/**
 * @param {string} s
 * @return {string}
 */
var reverseWords = function(s) {
    let str = s.split(" ");
    str = str.reverse();
    console.log(str);
    
    let ans = "";

    for(let i=0; i<str.length; i++){
        if(str[i].length==0){
            continue;
        }
        else{
            ans+=str[i];
            ans+=" ";
        }
    }

    ans = ans.slice(0,-1);

    return ans;
};